### PR TITLE
Add `List` component

### LINKED ISSUE #976: Quick-access tiles (list variation)
### Description
For an exemplar project, we need a new variation of a list component.

In this particular scenario the list is used to link to items (think recenty-opened-files).

This component has a heading, description and button. The entire list item should be clickable.

Please ensure the colour contrast matches accessibility guidelines - the example below has not been checked.

quick-access-tiles

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue
Closes #976 

## Overview
Adds a `List` component.

## Reason
Required for exemplar project.

## Work carried out
- [x] Add component
- [x] Add docs
- [x] Warn if overwriting props that are used for rendering

## Screenshot
Screenshot 2020-07-07 at 15 28 59

### CODE REVIEW

File: packages/css-framework/src/components/_list.scss [hunk 1]
```diff
@@ -0,0 +1,63 @@
+@use "../abstracts/functions" as f;
+
+.rn-list {
+  border: f.spacing("px") solid f.color("neutral", "100");
```
Comment (m7kvqbe1) [inline packages/css-framework/src/components/_list.scss:4]: I'm not familiar with this f.spacing `px` usage - what does this do?

File: packages/docs-site/src/library/pages/components/list.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+title: List
+description: todo
+header: true
+---
```
Comment (m7kvqbe1) [inline packages/docs-site/src/library/pages/components/list.md:6]: Worth adding `draft: true` to the frontmatter if the design docs are due to come at a later date.

File: packages/react-component-library/src/components/List/ListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <li className={classes} data-testid="list-item">
+      <button
+        className="rn-list__item-content"
+        onClick={onClick}
+        onMouseEnter={onMouseEnter}
+        onMouseLeave={onMouseLeave}
+      >
+        <div>
+          <h2 className="rn-list__item-title">{title}</h2>
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/List/ListItem.tsx:41]: Is `h2` the right tag? Maybe not a big deal - open to debate I think.

**Review by m7kvqbe1 — CHANGES_REQUESTED** [review]
LGTM - just a few tiny comments

File: packages/react-component-library/src/components/List/ListItem.tsx (hunk 3 above)
Comment (thyhjwb6) [reply packages/react-component-library/src/components/List/ListItem.tsx:41]: Having spoken with @hxltrhuxze we feel `h2` is fine at this stage.

File: packages/css-framework/src/components/_list.scss (hunk 1 above)
Comment (thyhjwb6) [reply packages/css-framework/src/components/_list.scss:4]: It is for `1px`:

> as everything else is set in rems having a single pixel value is often quite useful

**Review by m7kvqbe1 — APPROVED** [review]

**Pull request merged**